.github/pull_request_template.md

```diff
@@ -2,18 +2,28 @@
 
 ## Related issue number (if any)
 
-## Loom demo (if any)
-
-## Checklist before requesting a review
-- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
-- [ ] I have performed a self-review of my code.
-- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
-- [ ] If this is a new feature, I have added unit tests and integration tests.
-- [ ] I have run the integration tests locally and they are passing.
-- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
-- [ ] All features on the UI continue to work correctly.
-- [ ] Added one of the following CI labels:
-    - `run_integration_test`: Runs integration tests
-    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)
-
+## Test Plans
+Reviewer note: by approving the PR, reviewer agree that the test plans are solid enough to merge, with risk low enough for daily tests / release eng to take over.
+### Tests specific to this PR
+Include tests specific to this PR's context here:
+### Tests based on components touched by this PR
+Check the components touched by this PR and ensures all items under that component are checked. If certain item is not checked, explicitly state why and agree with PR reviewers.
+- [ ] This PR involves SDK changes
+  - [ ] Have integration test that covers my change
+  - [ ] Added 'run_integration_test' label to this PR
+  - [ ] If you are changing user-facing APIs, the change is backward compatible
+- [ ] This PR involves backend (go and executor changes)
+  - [ ] Have integration test that covers my change
+  - [ ] If changing REST endpoint, have backend test that covers this change (under `integration_tests/backend`)
+  - [ ] Added 'run_integration_test' label to this PR
+  - [ ] This PR involves changes in database layer
+    - [ ] There's a database unit test covering this change
+    - [ ] If the change affects how data is stored, has carefully examined if migration is required.
+- [ ] This PR involves engine-specific changes
+  - [ ] Performed integration tests (manually or automatically) backed by this engine.
+- [ ] This PR involves UI changes
+  - [ ] All manual_qa_tests workflows works with this feature.
```

Comment:
It's possible that the manual qa tests don't test this UI feature at all. Should we just add a general note at the bottom that if this is the case we should create a task? Curious what the process is for dealing with manual QA testing gaps.

Reply:
That's a good idea, we should ask PR owner to ask manual QA (or guideline) that can help cover the feature. I'd imagine in most cases it should be sufficient to modify an existing workflow